test/blocks/global-footer/global-footer.test.js

```diff
@@ -0,0 +1,247 @@
+        insertDummyElementOnTop({ height: window.innerHeight + 400 });
+
+        await createFullGlobalFooter({ waitForDecoration: false });
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(false));
+
+        clock.tick(3000);
+        await waitForFooterToDecorate();
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(true));
+      });
+
+      it('should render the footer when outside of the viewport, but within 300px range', async () => {
+        insertDummyElementOnTop({ height: window.innerHeight + 200 });
+        const startTime = performance.now();
+        await createFullGlobalFooter({ waitForDecoration: true });
+        const endTime = performance.now();
+        const timeDiff = endTime - startTime;
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(true));
+        // footer decoration should take less than 3s if within 300px range of viewport
+        expect(timeDiff < 3000).to.equal(true);
+      });
+
+      it('should render the footer when outside of the 300px viewport range, but scrolled into view earlier than 3s', async () => {
+        insertDummyElementOnTop({ height: window.innerHeight + 400 });
+        const startTime = performance.now();
+        await createFullGlobalFooter({ waitForDecoration: false });
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(false));
+
+        window.scrollBy(0, window.innerHeight);
+        await waitForFooterToDecorate();
+
+        Object.keys(allSelectors).forEach((key) => expect(
+          document.querySelector(allSelectors[key]) instanceof HTMLElement,
+        ).to.equal(true));
+
+        const endTime = performance.now();
+        const timeDiff = endTime - startTime;
+        // footer decoration should take less than 3s when scrolled into view
+        expect(timeDiff < 3000).to.equal(true);
+
+        expect(allElementsVisible(
+          visibleSelectorsDesktop,
+          document.querySelector(containerSelector),
+        )).to.equal(true);
+      });
+    });
+
+    describe('region picker tests', () => {
+      it('should handle non-empty hash', async () => {
+        await createFullGlobalFooter({ waitForDecoration: true });
+
+        const regionPickerElem = document.querySelector('.feds-regionPicker');
+        regionPickerElem.dispatchEvent(new Event('click'));
+
+        expect(regionPickerElem.getAttribute('href') === '#langnav').to.equal(true);
+        expect(regionPickerElem.hasAttribute('aria-expanded')).to.equal(true);
+        expect(regionPickerElem.getAttribute('aria-expanded') === 'true').to.equal(true);
```

Comment:
Could these not be grouped together and check `getAttribute('aria-expanded').to.equal('true')`? Same for L192-193

Reply:
Done 👍